File: src/diff.ts
import type { NetBytes } from './read'

export const diff_speed = (prev: NetBytes, curr: NetBytes) => {
    const dt = (curr.time - prev.time) / 1000

    return Object.freeze({
        rx: (curr.rx - prev.rx) / dt,
        tx: (curr.tx - prev.tx) / dt,
    })
}

File: src/diff.rs
use crate::{read::NetBytes, unit::Speed};

pub struct NetSpeed {
    pub rx: Speed,
    pub tx: Speed,
}

impl NetSpeed {
    pub fn diff_speed(prev: NetBytes, curr: NetBytes) -> Self {
        let dt = (curr.time - prev.time) as f64 / 1000.0;

        NetSpeed {
            rx: Speed::to_human_speed((curr.rx - prev.rx) as f64 / dt),
            tx: Speed::to_human_speed((curr.tx - prev.tx) as f64 / dt),
        }
    }
}
